tracking/include/pcl/tracking/impl/tracking.hpp

```diff
@@ -129,6 +129,30 @@ struct EIGEN_ALIGN16 ParticleXYZRPY : public _ParticleXYZRPY {
     return {trans_x, trans_y, trans_z, trans_roll, trans_pitch, trans_yaw};
   }
 
+  template <class InputIterator>
+  static ParticleXYZRPY
+  weightedAverage(InputIterator first, InputIterator last)
+  {
+    ParticleXYZRPY wa;
+    float wa_roll_sin = 0.0, wa_roll_cos = 0.0, wa_pitch_sin = 0.0, wa_pitch_cos = 0.0,
+          wa_yaw_sin = 0.0, wa_yaw_cos = 0.0;
+    for (auto point = first; point != last; point++) {
+      wa.x += point->x * point->weight;
+      wa.y += point->y * point->weight;
+      wa.z += point->z * point->weight;
+      wa_pitch_cos = std::cos(point->pitch);
+      wa_roll_sin += wa_pitch_cos * std::sin(point->roll) * point->weight;
+      wa_roll_cos += wa_pitch_cos * std::cos(point->roll) * point->weight;
+      wa_pitch_sin += std::sin(point->pitch) * point->weight;
+      wa_yaw_sin += wa_pitch_cos * std::sin(point->yaw) * point->weight;
+      wa_yaw_cos += wa_pitch_cos * std::cos(point->yaw) * point->weight;
+    }
+    wa.roll += std::atan2(wa_roll_sin, wa_roll_cos);
+    wa.pitch += std::asin(wa_pitch_sin);
+    wa.yaw += std::atan2(wa_yaw_sin, wa_yaw_cos);
```

Comment:
Is there a reason why you use `+=` instead of `=` here? You don't care about the previous values, you just want to assign the new values, right?

Reply:
Yes you're right, it doesn't make sense for it to be there, should be just assignment.